### ISSUE TITLE
1.7.0 completely breaks my vscode

### ISSUE THREAD

**Issue by Chippi** [buehler/typescript-hero#331]
<thank you for using and contributing to typescript hero>
<please fill the information below to ease up the process>
<you may even implement it and send a PR which would be very appreciated>

#### System information

- OS: Win 10 Pro / 64bit
- TypeScript Hero version: 1.7.0
- VSCode version: 1.17.2

#### Description
- Unable to type in any file
- GIT / Source control not working
- Extension host terminated unexpectedly.

![image](https://user-images.githubusercontent.com/6388933/32235801-bab7b7c8-be60-11e7-9e2c-1d2ca1b28af6.png)

**Comment by Dante-101** [comment]
The extension ends up using 2GB+ memory and crashes. 
I have few JS built by Webpack in my workspace. It always crashes with one of them in the stack trace.

```
[Extension Host] debugger inspector at %cFATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
%c
<--- Last few GCs --->

[7348:0000023CFFE53240]    37242 ms: Mark-sweep 2057.4 (2138.5) -> 2057.4 (2138.5) MB, 1192.9 / 0.0 ms  allocation failure GC in old space requested
[7348:0000023CFFE53240]    38488 ms: Mark-sweep 2057.4 (2138.5) -> 2057.3 (2122.5) MB, 1245.4 / 0.0 ms  last resort 
[7348:0000023CFFE53240]    39871 ms: Mark-sweep 2057.3 (2122.5) -> 2057.4 (2122.5) MB, 1381.7 / 0.0 ms  last resort 


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000000EBAB72B959 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [<readacted>\.vscode\extensions\rbbit.typescript-hero-1.7.0\node_modules\typescript-parser\TypescriptParser.js:~70] [pc=0000014239097FA9](this=0000037772C82311 <undefined>,o=000001FD8238AEE1 <String[91]: <readacted>\public-html\common\js\at-home\at-home.app.693f226ff88508f2515d.js>)
    2: arguments adaptor frame: 3->1
    3: map [...
```
Is there a way to make it typescript only?

**Comment by Dante-101** [comment]
I figured out that setting `typescriptHero.resolver.resolverMode` to `TypeScript` solves the out of memory crashing issue caused by parsing of JS files.

However, the issue of not being able to type in any file (delete/backspace works!) right after the boot up until the TSH resolver finishes still exists.

**Comment by buehler** [comment]
Holy.

**Comment by buehler** [comment]
This must be a big workspace (aka project) right?

**Comment by Dante-101** [comment]
It is not that big. ~15K lines of TS.

**Comment by Chippi** [comment]
598 ts files.  node_modules excluded.

**Comment by ylibrach** [comment]
Also having an issue since the 1.7 update. Fairly large project but didn't have issues prior to 1.7. Setting `typescriptHero.resolver.resolverMode` to TypeScript as suggested above seems to provide a reprieve for now.

**Comment by buehler** [comment]
Do you all use code-insiders build with multi-root workspaces? or just normal ones?

Since they build multi-root workspaces, I had to recode some of the indexing logic. Maybe that's the crux there.

**Comment by Dante-101** [comment]
I have the normal one. Haven't switched over to multi-root even once.

I think creating a web app project, where Webpack outputs to the project's folder/sub-folder should reproduce the issue. If it matters, I also have the source maps enabled for the output. Though I haven't tried it on a standalone small project, this is what is happening on my system.

**Comment by buehler** [comment]
The sad point is, when I use the extension (develop compiled version) I don't have any issues. But the actual codebase I'm working on is not that big.

I'll have a look and try to reduce the memory footprint. I assume the filter that ignores files is broken. Maybe I need to switch to globs so I don't even find files that are not supposed to be indexed.

**Comment by Dante-101** [comment]
For me, files that Webpack outputs were not excluded from the filter. If the extension can heuristically figure out that a given JS file coming from bundlers like Rollup, Webpack, etc., that will be great. That won't solve the issue fundamentally but hopefully, will not crash the extension.

@Chippi Can you tell us about the files which crash the extension? Does setting the resolver to Typescript solved your issue?

In my case, the memory footprint for Typescript only mode doesn't look bad. The issue looks more like an infinite loop with a memory leak somewhere. The CPU is consistently high, and the RAM usage grows linearly.

**Comment by buehler** [comment]
@Dante-101 The parser runs through the node tree recursively. So with a veeeery big file (like a webpack compiled result) it does run for a long time and eventually runs into the memory issues. So this could be the error. I don't know if I can heuristically find such files, but at least make some default assumptions like `build/**` `out/**` `dist/**`

**Comment by Dante-101** [comment]
Fair enough. I believe adding that to documentation will help.

Also, do you have any explanation for the issue of not being able to type in any file (delete/backspace works) right after the VS Code boot up until the TSH resolver finishes?

**Comment by Chippi** [comment]
@Dante-101 done some testing now. 

`"typescriptHero.resolver.resolverMode":"TypeScript"` seems to fix the crash issue. However, tons of cpu is being used on ts file open, but I guess It's necessary.

**Comment by Dante-101** [comment]
I opened up my fork of DefinitelyTyped and the extension crashed even with the above-mentioned settings.

```
[Extension Host] debugger inspector at %cFATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
%c
<--- Last few GCs --->

[21128:0000019A0CE93200]    63152 ms: Mark-sweep 2053.1 (2109.7) -> 2053.1 (2109.7) MB, 1069.3 / 0.0 ms  last resort 


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000000BCAB2AB959 <JS Object>
    2: createChildren [<redacted>\.vscode\extensions\rbbit.typescript-hero-1.7.0\node_modules\typescript\lib\typescript.js:~90652] [pc=00000054339B54BA](this=000001B308123F89 <a NodeObject with map 00000116AC645229>,sourceFile=000001DA7F782311 <undefined>)
    3: getChildren [<redacted>\.vscode\extensions\rbbit.typescript-hero-1.7.0\node_modules\typescript\lib\typescript.js:~90708] ...
```

Not sure if it matters, the error with code and signal:
`Extension host terminated unexpectedly. Code:  3  Signal:  null`

Any ideas?

**Comment by MadaraUchiha** [comment]
This still happens to me. Where can I download a 1.6.0 VSIX from to downgrade?

**Comment by buehler** [comment]
@Dante-101 most definitely because of the wrong excludes.
@MadaraUchiha uhm. Don't know actually. I'm working on it. Just need to write some tests for it, but then it should be ready to ship.

**Comment by MadaraUchiha** [comment]
For all interested, TSH 1.6.0 can be downloaded here and installed via "Install from VSIX": https://marketplace.visualstudio.com/_apis/public/gallery/publishers/rbbit/vsextensions/typescript-hero/1.6.0/vspackage

Make sure that auto updates are off. You need to uninstall 1.7.0 before you install 1.6.0

**Comment by ingvevs** [comment]
Perfect, downgrading to 1.6.0 brought my vscode back to life. Thanks for the tip!

**Comment by MadaraUchiha** [comment]
@buehler I don't understand, is this fixed? Is it safe to upgrade to 1.7.0 now? Do we need to wait for 1.8.0?

**Comment by Kleyguerth** [comment]
I had no problems on 1.7.0, but now I'm having this exact problem on 1.8.0.
